specs/vep-1493-vault-integration/README.md

```diff
@@ -258,19 +258,19 @@ Changes to the properties cli command:
 We are going to enhance the VDK-CS configuration with an optional Spring Vault Configuration.
 
 ```yaml
-spring.cloud.vault:
-    host: localhost
-    port: 8200
-    scheme: https
-    uri: https://localhost:8200
-    connection-timeout: 5000
-    read-timeout: 15000
-    config:
-    token: 19aefa97-cccc-bbbb-aaaa-225940e63d76
+# Hashicorp Vault Integration settings
+# When disabled/not configured the Secrets functionality won't work
+featureflag.vault.integration.enabled=true
+vdk.vault.uri=https://localhost:8200/v1/
+vdk.vault.approle.roleid=z2d59142-9b53-d163-1ae9-d6286d3dfe22
+vdk.vault.approle.secretid=c336f9e9-1555-8c29-9a7f-283608d06fbd
+datajobs.vault.size.limit.bytes=1048576
 ```
-The configuration can be marked optional as outlined in the
-[documentation](https://docs.spring.io/spring-cloud-vault/docs/current/reference/html/#vault.configdata.location.optional)
-which allows users who are not interested in using a secret storage, to simply disable this feature.
+The feature flag allows users who are not interested in using a secret storage, to simply disable this feature.
+
+The VDK Control Service is going to authenticate using Vault's [AppRole mechanism](https://developer.hashicorp.com/vault/tutorials/auth-methods/approle).
+THe Service Operator should setup the App Role authentication based on their needs in Vault, and provide the Vault URI
```

Comment:
Out of the box: no. There is a way to enable app roles and set them up with a couple of (literally) vault commands, however the resulting role and secret id's are randomized, so you'll still need to obtain them from vault and put them in the configuration. If you follow the linked guide - it takes less than 10 minutes to setup the whole thing using the vault CLI.

Reply:
I am asking because I am wondering if we should not keep token way for development purposes or quick getting started purposes. If I want to test the secrets API or try it out for something .  .

Reply:
OK... fixed... I've updated the application.properties file as well with instructions on how to start a local vault instance.